README.md

```diff
@@ -41,21 +42,29 @@ Follow these steps to open this sample in a container using the VS Code Dev Cont
 
 Once you have this sample opened, you'll be able to work with it like you would locally.
 
-> **Note:** This container runs as a non-root user with sudo access by default. Comment out `"remoteUser": "vscode"` in `.devcontainer/devcontainer.json` if you'd prefer to run as root.
-
 Some things to try:
 
 1. **Edit:**
    - Open `main.cpp`
    - Try adding some code and check out the language features.
-   - Notice that the C++ extension is already installed in the container since the `.devcontainer/devcontainer.json` lists `"ms-vscode.cpptools"` as an extension to install automatically when the container is created.
+   - Make a spelling mistake and notice it is detected. The [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) extension was automatically installed because it is referenced in `.devcontainer/devcontainer.json`.
+   - Also notice that utilities like `Vcpkg` and the [C++](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) extension are installed. Tools are installed in the `mcr.microsoft.com/devcontainers/cpp` image and Dev Container settings and metadata are automatically picked up from [image labels](https://containers.dev/implementors/reference/#labels).
+
 1. **Terminal:** Press <kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>\`</kbd> and type `uname` and other Linux commands from the terminal window.
+
 1. **Build, Run, and Debug:**
    - Open `main.cpp`
    - Add a breakpoint (e.g. on line 7).
    - Press <kbd>F5</kbd> to launch the app in the container.
    - Once the breakpoint is hit, try hovering over variables, examining locals, and more.
 
+1. **Install the GitHub CLI using a Dev Container Feature:**
+   - Press <kbd>F1</kbd> and select the **Dev Containers: Configure Container Features...** or **Codespaces: Configure Container Features...** command.
+   - Type "github" in the text box at the top.
+   - Check check box next to "GitHub CLI" (published by devcontainers) 
+   - Click OK
+   - Press <kbd>F1</kbd> and select the **Dev Containers: Rebuild Container** or **Codespaces: Rebuild Container** command so the modifications are picked up.
+

```

Comment:
This addition prompts an important question for me.

What are the purposes of this repository?
* To prove to a new-to-dev containers C++ user that the system works for a simple example?
* To expose C++ to existing dev containers users (providing them an idea of what a functioning C++ project should look like)?
* To provide a playground to learn about dev container features for someone who just happens to prefer/select the C++ example?

Reply:
The purpose of this repository is to demonstrate the benefits of Dev Containers to C++ users. The sample focuses on things you can do with Dev Containers as a result.  Key things to illustrate include:

1. The extension works in this context just like it does locally
2. Image metadata capabilities that allow dev container metadata previously required in devcontainer.json to be directly tied to the image. This avoids "drift" where the image and devcontainer.json get out of sync, enables mass updates, and simplifies devcontainer.json contents at a project level.
3. How easy it is to add additional capabilities using Dev Container Features

In some cases, extension authors have also opted to illustrate capabilities of the extension (see https://github.com/microsoft/vscode-remote-try-go).  Feel free to submit a PR with steps like this if you would like.